pkg/helpers/helpers.go

```diff
@@ -109,82 +110,124 @@ func GetMaxConcurrentReconciles() int {
 	return maxConcurrentReconciles
 }
 
-// GenerateClientFromSecret generate a client from a given secret
-func GenerateClientFromSecret(secret *corev1.Secret) (*ClientHolder, meta.RESTMapper, error) {
```

Comment:
the old `GenerateClientFromSecret` uses kubeconfig OR server/token, but after changing, the `GenerateImportClientFromKubeConfigSecret` will only use kubeconfig?

so the callers like `pkg/controller/clusterdeployment/clusterdeployment_controller.go` will be impacted?

Reply:
yes, handled by https://github.com/stolostron/managedcluster-import-controller/pull/302/files#diff-5fc238b39e2a1fc17bd412fbac2d14eab4e0f205c91886b5671611bec4c665ccR51

Reply:
> This because for clusterdeployment, it only have kubeconfig, right?

not sure about this, we need to confirm. Or can we just keep the origin logic? read kubeconfig first, if it does not exist, read token.

Reply:
we read the kubeconfig from `clusterdeployment.spec.clusterMetadata.adminKubeconfigSecretRef`, it  should only have kubeconfig, https://github.com/openshift/hive/blob/master/apis/hive/v1/clusterprovision_types.go#L44`